### ISSUE TITLE
[java] UnnecessaryImport FP with annotation

### ISSUE THREAD

**Issue by tmzk1005** [pmd/pmd#3700]
**Affects PMD Version:** 

org.apache.maven.plugins:maven-pmd-plugin:3.15.0

pmd version  6.41.0

**Description:**

An used import is reported as unused import

**Exception Stacktrace:**

No Exception Stacktrace

**Code Sample demonstrating the issue:**

```java
import java.util.Optional;

import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.PagingAndSortingRepository;

import zk.apigw.common.api.ApiStatus;
import zk.apigw.manager.bean.entity.Api;

public interface ApiRepository extends PagingAndSortingRepository<Api, Long> {

    long countByStatus(ApiStatus apiStatus);

    Optional<Api> findByIdAndStatus(Long id, ApiStatus status);

    @Query("select id from zk.apigw.manager.bean.entity.Api api where api.status = zk.apigw.common.api.ApiStatus.PUBLISHED")
    Iterable<Long> findAllPublishedApiIds();

}
```

A JPA repository interface has a method with `@Query` annotation. when run ''mvn package", pmd check failed: `Unused import 'org.springframework.data.jpa.repository.Query'`

**Steps to reproduce:**

1. enable rule: <rule ref="category/java/bestpractices.xml/UnusedImports"/>
2. config org.apache.maven.plugins:maven-pmd-plugin
3. run `mvn package`

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

Maven

**Comment by AbuBakrCh** [comment]
Hi @tmzk1005 , should not the title be: "**A used import** is reported as unused import

**Comment by adangel** [comment]
Hi @tmzk1005 ,
I can't reproduce the problem. See https://github.com/adangel/scratchpad/tree/master/pmd-issue-3700-unnecessaryimports-annotations for a minimal test case.
Do you see, what you do differently?

**Comment by tmzk1005** [comment]
Sorry, maybe it is not a problem for the latest version.

I thought I wsd using pmd version 6.41.0, but actually using 6.3x.x. After I upgrade pmd plugin, no UnnecessaryImport problem happen.

**Comment by oowekyala** [comment]
Thanks for the feedback, I'm closing this then
